src/libLookup/Lookup.cpp

```diff
@@ -490,14 +490,15 @@ void Lookup::SendMessageToRandomLookupNode(const bytes& message) const {
     return;
   }
 
-  // To avoid sending message to multiplier
+  // To avoid sending message to multiplier and himself
   VectorOfNode tmp;
   std::copy_if(m_lookupNodes.begin(), m_lookupNodes.end(),
                std::back_inserter(tmp), [this](const PairOfNode& node) {
-                 return find_if(m_multipliers.begin(), m_multipliers.end(),
-                                [&node](const PairOfNode& mult) {
-                                  return node.second == mult.second;
-                                }) == m_multipliers.end();
+                 return (find_if(m_multipliers.begin(), m_multipliers.end(),
+                                 [&node](const PairOfNode& mult) {
+                                   return node.second == mult.second;
```

Comment:
The code logic is correct but so hard to read, can implement in a more readable way?

Reply:
Can't think of better readable way than this together with preserving performance ? Please suggest.
Note: added comment though for readability of logic.